macros/src/newtype.rs

```diff
@@ -0,0 +1,82 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, Data, DeriveInput, Fields};
+
+pub fn cw_storage_newtype_key_derive(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
```

Comment:
I wonder if you can parse it directly as ItemStruct (or whatever the type is called in `syn`). Could save some of the pattern matching later.

Reply:
yup! done in https://github.com/CosmWasm/cw-storage-plus/pull/88/commits/6a1c6d1caae918dcc781ec1596696d32b8d18d6b